src/Phinx/Db/Table/Index.php

```diff
@@ -59,12 +59,12 @@ class Index
     /**
      * Sets the index columns.
      *
-     * @param string[] $columns Columns
+     * @param string|string[] $columns Columns
      * @return $this
      */
-    public function setColumns($columns)
+    public function setColumns($columns): Index
     {
-        $this->columns = $columns;
+        $this->columns = is_string($columns) ? [$columns] : $columns;
```

Comment:
This change makes this method equivalent to how `setColumns` works in the `ForeignKey` class.

Reply:
Though the handling of the `$columns` value is very inconsistent across the codebase. A number of the `AlterInstruction` functions within the database adapters accepts a `string $column` property while some accept a `Column $column`, and then elsewhere we have `string|string[] $columns` or just `string[] $columns`. A follow-up PR would perhaps be good to work on making this more consistent throughout.